File: Data Structures & Algorithms/insertionSort/submission-0.py
# Definition for a pair.
# class Pair:
#     def __init__(self, key: int, value: str):
#         self.key = key
#         self.value = value
class Solution:
    def swap(self, ls, i, j):
        tmp = ls[i]
        ls[i] = ls[j]
        ls[j] = tmp

    def insertionSort(self, pairs: List[Pair]) -> List[List[Pair]]:
        states = []
        n = len(pairs)
        
        ls = pairs.copy()
        for i in range(n):
            for j in reversed(range(i)):
                if ls[j + 1].key < ls[j].key:
                    self.swap(ls, j, j + 1)
                else:
                    break

            states.append(ls.copy())

        return states

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int i, j, tmp;

        for (i = 0, j = numbers.size() - 1; i < j; ) {
            tmp = numbers[i] + numbers[j];

            if (tmp == target)
                return {i + 1, j + 1};
            else if (tmp < target)
                i++;
            else
                j--;
        }

        return {-1, -1};
    }
};
